tests/WE2E/run_WE2E_tests.py

```diff
@@ -0,0 +1,447 @@
+            logging.debug(test_cfg['task_get_extrn_lbcs'])
+
+
+        logging.debug(f"Writing updated config.yaml for test {test_name}\nbased on specified command-line arguments:\n")
+        logging.debug(cfg_to_yaml_str(test_cfg))
+        with open(ushdir + "/config.yaml","w") as f:
+            f.writelines(cfg_to_yaml_str(test_cfg))
+
+        logging.debug(f"Calling workflow generation function for test {test_name}\n")
+        if args.quiet:
+            console_handler = logging.getLogger().handlers[1]
+            console_handler.setLevel(logging.WARNING)
+        expt_dir = generate_FV3LAM_wflow(ushdir,logfile=f"{ushdir}/log.generate_FV3LAM_wflow",debug=args.debug)
+        if args.quiet:
+            if args.debug:
+                console_handler.setLevel(logging.DEBUG)
+            else:
+                console_handler.setLevel(logging.INFO)
+        # If this job is not using crontab, we need to add an entry to monitor.yaml
+        if 'USE_CRON_TO_RELAUNCH' not in test_cfg['workflow']:
+            test_cfg['workflow'].update({"USE_CRON_TO_RELAUNCH": False})
+        if not test_cfg['workflow']['USE_CRON_TO_RELAUNCH']:
+            logging.debug(f'Creating entry for job {test_name} in job monitoring dict')
+            monitor_yaml[test_name] = dict()
+            monitor_yaml[test_name].update({"expt_dir": expt_dir})
+            monitor_yaml[test_name].update({"status": "CREATED"})
+
+    if not args.use_cron_to_relaunch:
+        logging.info("calling function that monitors jobs, prints summary")
+        monitor_file = monitor_jobs(monitor_yaml, args.debug)
+
+        logging.info("All experiments are complete")
+        logging.info(f"Summary of results available in {monitor_file}")
+
+
+
+
+
+def check_tests(tests: list) -> list:
+    """
+    Function for checking that all tests in a provided list of tests are valid
+
+    Args:
+        tests        : List of potentially valid test names
+    Returns:
+        tests_to_run : List of config files corresponding to test names
+    """
+
+    testfiles = glob.glob('test_configs/**/config*.yaml', recursive=True)
+    tests_to_run=[]
+    for test in tests:
+        # Skip blank/empty testnames; this avoids failure if newlines or spaces are included
+        if test.isspace():
+            continue
+        match=False
+        # Search for exact config file name to avoid accidental partial matches
+        test_config='config.' + test.rstrip() + '.yaml'
+        for testfile in testfiles:
+            if test_config in testfile:
+                logging.debug(f"found test {test}")
+                match=True
+                tests_to_run.append(os.path.abspath(testfile))
+        if not match:
+            raise Exception(f"Could not find test {test}")
+    # Because some test files are symlinks to other tests, check that we don't
+    # include the same test twice
```

Comment:
@mkavulich I thought you were getting rid of the symlinks feature.  Seems like you're still checking for that here.

Reply:
Sorry if I was confusing in our previous conversation; I had no plans to change the existence of symlinks; I only changed it so that this does not result in a failure, it simply omits duplicate tests. Whether or not that's the best approach, I'm all ears on opinions there.